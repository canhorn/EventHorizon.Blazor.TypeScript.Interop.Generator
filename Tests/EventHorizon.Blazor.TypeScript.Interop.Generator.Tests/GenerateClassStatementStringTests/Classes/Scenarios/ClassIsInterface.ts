declare module Examples {
  export interface ExampleClass<T, N extends number> {
    method(): ExampleClass<T, N>;
  }
}
