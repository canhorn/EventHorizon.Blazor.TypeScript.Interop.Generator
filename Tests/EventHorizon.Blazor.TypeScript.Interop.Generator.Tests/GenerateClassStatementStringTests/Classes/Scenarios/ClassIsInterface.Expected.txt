/// Generated - Do Not Edit
namespace Examples
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    public interface ExampleClass<T, N> : ICachedEntity { }
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClassCachedEntity<CachedEntity>>))]
    public class ExampleClassCachedEntity<T, N> : CachedEntityObject, ExampleClass<T, N> where T : CachedEntity, new() where N : CachedEntity, new()
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties

        #endregion
        
        #region Constructor
        public ExampleClassCachedEntity() : base() { }

        public ExampleClassCachedEntity(
            ICachedEntity entity
        ) : base(entity)
        {
        }


        #endregion

        #region Methods
        public ExampleClassCachedEntity<T, N> method()
        {
            return EventHorizonBlazorInterop.FuncClass<ExampleClassCachedEntity<T, N>>(
                entity => new ExampleClassCachedEntity<T, N>() { ___guid = entity.___guid },
                new object[]
                {
                    new string[] { this.___guid, "method" }
                }
            );
        }
        #endregion
    }
}